core/src/main/scala/org/apache/spark/sql/delta/stats/DataSkippingReader.scala

```diff
@@ -416,7 +416,8 @@ trait DataSkippingReaderBase
       constructDataFilters(IsNull(e))
 
     // Match any file whose min/max range contains the requested point.
-    case EqualTo(SkippingEligibleColumn(a), SkippingEligibleLiteral(v)) =>
+    case Equality(SkippingEligibleColumn(a), lit @ SkippingEligibleLiteral(v)) =>
+      assert(lit.value != null)
```

Comment:
nit: do we still need this?

Reply:
It's just for the sanity check, to make sure the optimization rules not be changed by Spark.

Reply:
QQ: Spark allows `Literal(null, StringType)`, although I'm not sure in which case this could happen.

If `Literal(null, StringType)` is possible, `c1 = Literal(null, StringType)` is still correct but `c1 <=> Literal(null, StringType)` would be wrong because the `DataSkippingPredicate` below may drop `null` values.

In addition, it would be great to construct unit tests for such case.

Reply:
Since we do not know when `c1 <=> Literal(null, StringType)` will happen, I think we can return `None` when the input expression is `c1 <=> Literal(null, StringType)`, WDYT?

Reply:
the unit tests for such case will be added later.

Reply:
@zsxwing updated.